hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -213,7 +214,24 @@ class MergeOnReadSnapshotRelation(sqlContext: SQLContext,
         PartitionedFile(getPartitionColumnsAsInternalRow(file.getFileStatus), filePath, 0, file.getFileLen)
       }
 
-      HoodieMergeOnReadFileSplit(partitionedBaseFile, logFiles)
+      if (shouldExtractPartitionValuesFromPartitionPath && logFiles.nonEmpty) {
```

Comment:
I don't think we actually need this -- log-files always co-located w/ base-file and as such we can extract partition-values from there